File: scrape.py
import urllib.request
import re
import json

req = urllib.request.Request('https://www.books.com.tw/web/sys_saletopb/books', headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
html = urllib.request.urlopen(req).read().decode('utf-8')
urls = set(re.findall(r'https://im[12]\.book\.com\.tw/image/getImage\?i=[^&\"\' ]+&v=[^&\"\' ]+', html))
urls = [u.replace('&amp;', '&') + "&w=400&h=600" for u in urls if 'image/getImage' in u]
print(json.dumps(urls[:100], indent=2))

File: scrape.js
const https = require('https');

const options = {
  hostname: 'www.books.com.tw',
  path: '/web/sys_saletopb/books',
  method: 'GET',
  headers: {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36'
  }
};

const req = https.request(options, (res) => {
  let data = '';
  res.on('data', (chunk) => { data += chunk; });
  res.on('end', () => {
    const regex = /https:\/\/im[12]\.book\.com\.tw\/image\/getImage\?i=([^&"']+)&v=([^&"']+)/g;
    const urls = new Set();
    let match;
    while ((match = regex.exec(data)) !== null) {
      if (match[0].includes('image/getImage')) {
        urls.add(match[0].replace(/&amp;/g, '&') + '&w=400&h=600');
      }
    }
    console.log(JSON.stringify(Array.from(urls).slice(0, 100), null, 2));
  });
});

req.on('error', (e) => {
  console.error(`Problem with request: ${e.message}`);
});

req.end();
